File: app/src/main/java/de/trackcovidcluster/server/SendJsonToServer.java
package de.trackcovidcluster.server;

import android.os.AsyncTask;
import android.util.Log;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.io.Writer;
import java.net.HttpURLConnection;
import java.net.URL;

import static android.content.ContentValues.TAG;

class SendJsonToServer extends AsyncTask<String,String,String> {

    public SendJsonToServer() {}

    @Override
    protected String doInBackground(String... params) {

        String JsonResponse = null;
        String JsonDATA = params[0];
        HttpURLConnection urlConnection;
        urlConnection = null;
        BufferedReader reader = null;

        try {
            URL url = new URL("https://api.trackcovidcluster.de:12345/json");
            urlConnection = (HttpURLConnection) url.openConnection();
            urlConnection.setDoOutput(true);

            // is output buffer writter
            urlConnection.setRequestMethod("POST");
            urlConnection.setRequestProperty("Content-Type", "application/json");
            urlConnection.setRequestProperty("Accept", "application/json");

            //set headers and method
            Writer writer = new BufferedWriter(new OutputStreamWriter(urlConnection.getOutputStream(), "UTF-8"));
            writer.write(JsonDATA);

            // json data
            writer.close();
            InputStream inputStream = urlConnection.getInputStream();

            //input stream
            StringBuffer buffer = new StringBuffer();

            if (inputStream == null) {
                // Nothing to do.
                return null;
            }

            reader = new BufferedReader(new InputStreamReader(inputStream));

            String inputLine;

            while ((inputLine = reader.readLine()) != null)
                buffer.append(inputLine + "\n");
            if (buffer.length() == 0) {
                // Stream was empty. No point in parsing.
                return null;
            }

            JsonResponse = buffer.toString();

            //response data
            Log.d("","_________________RESPONSE________________");
            Log.i(TAG,JsonResponse);

            try {
                //send to post execute
                return JsonResponse;
            } catch (Exception e) {
                e.printStackTrace();
            }

            return null;

        } catch (IOException e) {
            e.printStackTrace();
        }
        finally {
            if (urlConnection != null) {
                urlConnection.disconnect();
            }
            if (reader != null) {
                try {
                    reader.close();
                } catch (final IOException e) {
                    Log.e(TAG, "Error closing stream", e);
                }
            }
        }
        return null;
    }


    @Override
    protected void onPostExecute(String s) {

    }
}

File: app/src/main/java/de/trackcovidcluster/server/SendJsonToServer.kt
package de.trackcovidcluster.server

import android.content.ContentValues
import android.os.AsyncTask
import android.util.Log
import java.io.*
import java.net.HttpURLConnection
import java.net.URL

internal class SendJsonToServer : AsyncTask<String?, String?, String?>() {

    public override fun doInBackground(vararg params: String?): String? {
        var JsonResponse: String? = null
        val JsonDATA = params[0]
        var urlConnection: HttpURLConnection?
        urlConnection = null
        var reader: BufferedReader? = null
        try {
            val url = URL("https://api.trackcovidcluster.de:12345/json")
            urlConnection = url.openConnection() as HttpURLConnection
            urlConnection.doOutput = true

            // is output buffer writter
            urlConnection!!.requestMethod = "POST"
            urlConnection.setRequestProperty("Content-Type", "application/json")
            urlConnection.setRequestProperty("Accept", "application/json")

            //set headers and method
            val writer: Writer = BufferedWriter(
                OutputStreamWriter(
                    urlConnection.outputStream,
                    "UTF-8"
                )
            )
            writer.write(JsonDATA)

            // json data
            writer.close()
            val inputStream = urlConnection.inputStream

            //input stream
            val buffer = StringBuffer()
            if (inputStream == null) {
                // Nothing to do.
                return null
            }
            reader = BufferedReader(InputStreamReader(inputStream))
            var inputLine: String
            while (reader.readLine()
                    .also { inputLine = it } != null
            ) buffer.append(
                """
    $inputLine

    """.trimIndent()
            )
            if (buffer.length == 0) {
                // Stream was empty. No point in parsing.
                return null
            }
            JsonResponse = buffer.toString()

            //response data
            Log.d("", "_________________RESPONSE________________")
            Log.i(ContentValues.TAG, JsonResponse)
            try {
                //send to post execute
                return JsonResponse
            } catch (e: Exception) {
                e.printStackTrace()
            }
            return null
        } catch (e: IOException) {
            e.printStackTrace()
        } finally {
            urlConnection?.disconnect()
            if (reader != null) {
                try {
                    reader.close()
                } catch (e: IOException) {
                    Log.e(ContentValues.TAG, "Error closing stream", e)
                }
            }
        }
        return null
    }

    override fun onPostExecute(s: String?) {}
}